lib/supavisor/db_handler.ex

```diff
@@ -50,26 +50,28 @@ defmodule Supavisor.DbHandler do
   def handle_event(:internal, _, :connect, %{auth: auth} = data) do
     Logger.info("Try to connect to DB")
 
-    socket_opts = [
+    sock_opts = [
       :binary,
       {:packet, :raw},
-      {:active, true},
+      {:active, false},
       auth.ip_version
     ]
 
-    case :gen_tcp.connect(auth.host, auth.port, socket_opts) do
-      {:ok, socket} ->
+    case :gen_tcp.connect(auth.host, auth.port, sock_opts) do
+      {:ok, sock} ->
         Logger.debug("auth #{inspect(auth, pretty: true)}")
+        sock = {:gen_tcp, sock}
 
-        msg =
-          :pgo_protocol.encode_startup_message([
-            {"user", auth.user},
-            {"database", auth.database},
-            {"application_name", auth.application_name}
-          ])
+        case handshake(sock, auth) do
+          {:ok, sock} ->
+            # Back to active from here
+            set_sock_opts(sock, [{:active, true}])
+            {:next_state, :authentication, %{data | sock: sock}}
 
-        :ok = :gen_tcp.send(socket, msg)
-        {:next_state, :authentication, %{data | socket: socket}}
+          {:error, error} ->
+            Logger.error("Handshake error #{inspect(error)}")
+            {:keep_state_and_data, {:state_timeout, 2_500, :connect}}
```

Comment:
consider adding a `@some_timeout 2500` at the top of the module, otherwise it's easy to miss imo.